### ISSUE TITLE
Compile error

### ISSUE THREAD

**Issue by RH-Song** [Durant35/LatexThesis4SYSU#6]
I use ubuntu 16.04 and TexStudio.

When I compile pre.tex, it works smoothly.

However, when I compile the main.tex with pdflatex,  some errors occur.
![bianyi](https://user-images.githubusercontent.com/17057430/52185595-08211980-285c-11e9-861d-d4b092ea4e01.png)

- log:
Option clash for package hyperref. \usepackage
Critical xeCJK error: "Require-XeTeX" The xeCJK package requires XeTeX to function. You must change your typesetting engine to "xelatex" instead of plain "latex" or "pdflatex" or "lualatex". Loading xeCJK will abort! For immediate help type H <return>.
Undefined control sequence. \setmainfont
Missing \begin{document}. \setmainfont{T
Undefined control sequence. \setCJKmainfont
Undefined control sequence. \setCJKfamilyfont
Undefined control sequence. \setCJKfamilyfont
Missing $ inserted. \setCJKfamilyfont{fs}{FangSong_
Extra }, or forgotten $. \setCJKfamilyfont{fs}{FangSong_GB2312}
Undefined control sequence. \setCJKfamilyfont
Undefined control sequence. \setCJKfamilyfont
Undefined control sequence. \setCJKfamilyfont
Missing $ inserted.
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Argument of extract@rangefontinfo has an extra }. \makecover
Paragraph ended before extract@rangefontinfo was complete. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
Undefined control sequence. \makecover
fontenc already loaded - some fonts may be unprocessed.

**Comment by RH-Song** [comment]
I am sure that I have already installed windows-fonts in ubuntu.

**Comment by RH-Song** [comment]
When I did what the TexStudio say, changing the typeset engine to xelatex, I got other errors.

The log is show as follow:
fontspec error: "font-not-found" The font "WenQuanYi Zen Hei" cannot be found. For immediate help type H <return>. }
fontspec error: "font-not-found" The font "WenQuanYi Zen Hei" cannot be found. For immediate help type H <return>. }
Cannot use \XeTeXOTcountfeatures with nullfont; not an OpenType Layout font. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }
Missing = inserted for \ifnum. }
Cannot use \XeTeXOTfeaturetag with nullfont; not an OpenType Layout font. }

**Comment by Durant35** [comment]
1. Recommend to use **TexStudio** with **xeLaTex** rendering engine
2. I think it's lack of fonts, maybe
    + need to reboot???
    + try again, referring to my fonts' zip: https://github.com/Durant35/LatexThesis4SYSU/wiki/Ubuntu16.04

**Comment by RH-Song** [comment]
I fixed this problem by changing :
```
"\documentclass[fontset=ubuntu,UTF8,12pt,openany,oneside]{book}" 
```
into 
```
"\documentclass[UTF8,12pt,openany,oneside]{book}". 
```
That is to say, I remove "fontset=ubuntu".

I searched on the Internet and found that "WenQuanYi Zen Hei" is one of the default fonts in Ubuntu. People use it to replace "Zen Hei" which is a windows' font. Since it is inconvenient to set fonts when you use ctex font, Leo Liu, Qing Lee and Liam Huang contributed a script named "ctex-fontset-ubuntu.def" to textlive to set "WenQuanYi Zen Hei" as default font to replace "Zen Hei". 

However, I don't know why my ubuntu 16.04 didn't have "WenQuanYi Zen Hei".  So when using fontset=ubuntu which means using "WenQuanYi Zen Hei" to replace "Zen Hei", errors occur.

Considering that I have install windows fonts in my ubuntu sysytem, I removed "fontse=ubuntu". And it works.
